src/erlcloud_s3.erl

```diff
@@ -467,9 +468,17 @@ extract_acl(ACL) ->
     [extract_grant(Item) || Item <- ACL].
 
 extract_grant(Node) ->
-    [{grantee, extract_user(xmerl_xpath:string("Grantee", Node))},
+    [{grantee, extract_grantee(hd(xmerl_xpath:string("Grantee", Node)))},
      {permission, decode_permission(erlcloud_xml:get_text("Permission", Node))}].
 
+extract_grantee(Node) ->
```

Comment:
use of hd() does not guarantee you the list has only one element. Writing it as above will also check that that there only one element. and the reason of use of Lists is `xmerl_xpath`